Source/WebGPU/WGSL/TypeStore.cpp

```diff
@@ -34,6 +34,44 @@ namespace WGSL {
 
 using namespace Types;
 
+struct VectorKey {
```

Comment:
Do we need these keys? What not implement a hasher for `Type`?

Reply:
We're trying to map e.g. `T + 2` to `vec2<T>`, so just hashing the type wouldn't give us enough information.